plugins/kubernetes-backend/src/service/KubernetesBuilder.ts

```diff
@@ -76,12 +82,15 @@ export class KubernetesBuilder {
   private objectsProvider?: KubernetesObjectsProvider;
   private fetcher?: KubernetesFetcher;
   private serviceLocator?: KubernetesServiceLocator;
+  private proxy: KubernetesProxy;
 
   static createBuilder(env: KubernetesEnvironment) {
     return new KubernetesBuilder(env);
   }
 
-  constructor(protected readonly env: KubernetesEnvironment) {}
+  constructor(protected readonly env: KubernetesEnvironment) {
+    this.proxy = new KubernetesProxy(env.logger);
```

Comment:
All of the other things (locators etc) have a setter in the builder (e.g. `setServiceLocator`) so you can supply a custom one. Would it make sense to make this one configurable too instead of hardcoding the construction of it inside this constructor?

Reply:
As far as the variables we see in the project, the builder and it’s setters are used immediately, does it makes sense for us to just remove the setters and just use the constructor directly, do you have an example of a place where the instance of the builder and the setter functions are not chained? @freben

Reply:
@RubenV-dev If I'm understanding your question properly, are you suggesting that `KubernetesBuilder` be refactored and all of the setter functions be removed? If so, I think the answer is "no" -- as an example, [the docs](https://backstage.io/docs/features/kubernetes/installation#custom-cluster-discovery) describe using the `setClusterSupplier` method as a supported way to customize the behaviour of this plugin. From discussions I've had with @mclarke47, I'm pretty confident there are important use cases where `setServiceLocator` is used in the wild as well.

Please let me know if I've misunderstood something here!

Reply:
Since we used setters before for the other things, this could just be made into a new setter as well. Sure we could refactor into making it an options object but that'd be a separate effort and PR because it's a discussion unto itself.

Either way all of this will be moot once the backend system work falls into place, because at that point it all will be replaced by a sort of dependency injection and adaptation patterns thing. So let's stick to the current shape of things for now.

Reply:
Changed to setter/getter and chained properly in the build functions